solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -491,22 +490,27 @@ public long getIndexSize() {
   }
 
   public long getCachedIndexSize() {
-    if (indexSize.get() == null) {
-      if (log.isDebugEnabled()) {
-        log.debug("Recalculating index size for {}", getName());
+    SolrRequestInfo requestInfo = SolrRequestInfo.getRequestInfo();
+    if (requestInfo != null) {
+      Map<Object,Object> contextMap = requestInfo.getReq().getContext();
+      Long cachedIndexSize = (Long)contextMap.get(cachedIndexSizeKeyName(getName()));
```

Comment:
You could use the computeIfAbsent pattern which is more idiomatic than this older Java style?  Granted you'd probably lose the log for re-using a previous index size but I don't think the log is important.